src/math/lp/lar_core_solver_def.h

```diff
@@ -179,11 +179,17 @@ void lar_core_solver::solve() {
         }
         lp_assert(!settings().use_tableau() || r_basis_is_OK());
     }
-    if (m_r_solver.get_status() == lp_status::INFEASIBLE) {
+    switch (m_r_solver.get_status())
+    {
+    case lp_status::INFEASIBLE:
         fill_not_improvable_zero_sum();
-    } 
-    else if (m_r_solver.get_status() != lp_status::UNBOUNDED) {
+        break;
+    case lp_status::CANCELLED:
+    case lp_status::UNBOUNDED: // do nothing in these cases
```

Comment:
theory_lra only checks against lp_status::OPTIMAL when deciding whether to re-check feasiblity in final_check_eh. Now UNBOUNDED would likely leak too. So shouldn't theory_lra condition the status on being feasible, where feasible is one of OPTIMAL, UNBOUNDED or so?

Reply:
Yes, it seems the right approach. I can fix it later.

Reply:
The question is how theory_lra handles the return codes.

Can we always assume that if the status is OPTIMAL, FEASIBLE, UNBOUNDED that all bounds are satisfied?

If so, then there are two places to change in theory_lra. It makes no assumptions about feasibility with UNBOUNDED.

Reply:
Yes. Or, lp can have a boolean function, smth like, feasible_status(lp_status), providing this abstraction.

Reply:
yes please